File: apps/mybookkeeper/frontend/src/shared/types/rent/rent-period-summary.ts
import type { RentChargeStatus } from "./rent-charge-status";

/** The "how much has been paid for this period" answer, precomputed. */
export interface RentPeriodSummary {
  charge_id: string;
  label: string;
  period_start: string;
  period_end: string;
  amount: string;
  allocated: string;
  remaining: string;
  status: RentChargeStatus;
}

File: apps/mybookkeeper/backend/app/schemas/rent/rent_period_summary.py
"""The 'how much has been paid for this period' answer, precomputed."""
from __future__ import annotations

import datetime as _dt
import uuid
from decimal import Decimal

from pydantic import BaseModel


class RentPeriodSummary(BaseModel):
    charge_id: uuid.UUID
    label: str
    period_start: _dt.date
    period_end: _dt.date
    amount: Decimal
    # Set only when ``amount`` is a prorated slice of a part-month tenancy;
    # see ``RentChargeResponse.full_amount``.
    full_amount: Decimal | None = None
    allocated: Decimal
    remaining: Decimal
    status: str
